db/silkworm/stagedsync/stagedsync.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+   Copyright 2020 - 2021 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+           http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+#include <silkworm/db/tables.hpp>
+#include <silkworm/db/chaindb.hpp>
+#include <silkworm/stagedsync/util.hpp>
+#include <vector>
+
+#ifndef SILKWORM_STAGEDSYNC_HPP_
+#define SILKWORM_STAGEDSYNC_HPP_
+
+namespace silkworm::stagedsync {
+
+extern bool write_receipts;
+constexpr size_t kBatchSize = 512 * kMebi;
+
+typedef StageResult (*StageFunc)(lmdb::DatabaseConfig);
+typedef StageResult (*UnwindFunc)(lmdb::DatabaseConfig, uint64_t);
+
+struct Stage {
+    StageFunc stage_func;
+    UnwindFunc unwind_func;
+    uint64_t id;
+};
+
+// Stage functions
+StageResult stage_headers(lmdb::DatabaseConfig);
+StageResult stage_blockhashes(lmdb::DatabaseConfig);
+StageResult stage_bodies(lmdb::DatabaseConfig);
+StageResult stage_senders(lmdb::DatabaseConfig);
+StageResult stage_execution(lmdb::DatabaseConfig);
+/* HashState Promotion Functions*/
+
+/*
+    * Operation is used to distinguish what bucket we want to generated
+    * HashAccount is for genenerating HashedAccountBucket
+    * HashStorage is for genenerating HashedStorageBucket
+    * Code generates hashed key => code_hash mapping
+
+*/
+enum class HashstateOperation {
```

Comment:
`HashstateOperation` shouldn't be in `db/silkworm/stagedsync/stagedsync.hpp` since it's not something pertaining to all stages, but specific for a particular stage.

Reply:
I would like to be able to use promotions in the cmd freely tool in the cmd directory in case we need to test them in the future,  so that's why i kept cmd/hashstate the same and did not use stage_hashstate function there. as you can see you need to specify an Operation in promote